src/renderer/components/+preferences/select-metrics-dialog.tsx

```diff
@@ -0,0 +1,120 @@
+  Node = "Node",
+  Pod = "Pod",
+  Deployment = "Deployment",
+  StatefulSet = "StatefulSet",
+  Container = "Container",
+  Ingress = "Ingress",
+  VolumeClaim = "VolumeClaim",
+  ReplicaSet = "ReplicaSet",
+  DaemonSet = "DaemonSet",
+}
+
+const Placeholder = observer((props: PlaceholderProps<any>) => {
+  const getPlaceholder = (): React.ReactNode => {
+    if (userStore.hiddenMetrics.size >= 1) {
+      return <>Metrics: {Array.from(userStore.hiddenMetrics).join(", ")}</>;
+    }
+    else {
+      return <>Select metrics...</>;
+    }
+  };
+
+  return (
+    <components.Placeholder {...props}>
+      {getPlaceholder()}
+    </components.Placeholder>
+  );
+});
+
+@observer
+export class SelectMetricsDialog extends React.Component<Props> {
+
+  @observable static isOpen = false;
+
+  static open() {
+    SelectMetricsDialog.isOpen = true;
+  }
+
+  static close() {
+    SelectMetricsDialog.isOpen = false;
+  }
+
+  onChange(values: ResourceType[]) {
+    values.map(value => {
+      if (userStore.hiddenMetrics.has(value)) {
+        userStore.hiddenMetrics.delete(value);
+      } else {
+        userStore.hiddenMetrics.add(value);
+      }
+    });
+  }
+
+  reset = () => {
+    userStore.hiddenMetrics.clear();
+    userStore.preferences.hideMetrics = false;
+  };
+
+  render() {
+    const { ...dialogProps } = this.props;
+    const header = <h5>Select metrics to hide</h5>;
+
+    return (
+      <Dialog
+        {...dialogProps}
+        isOpen={SelectMetricsDialog.isOpen}
+        close={SelectMetricsDialog.close}
+      >
+        <Wizard header={header} done={SelectMetricsDialog.close}>
+          <WizardStep contentClass="flow column" nextLabel={"Reset"} next={this.reset} prevLabel={"Close"}>
+            <div>
+              <Select
+                className={"SelectMetricsDialog"}
+                components={{ Placeholder }}
+                isMulti
+                closeMenuOnSelect={false}
+                isDisabled={userStore.preferences.hideMetrics}
+                options={Object.values(ResourceType)}
+                onChange={(options: SelectOption[]) => {
+                  this.onChange(options.map(option => option.value));
+                }}
+                formatOptionLabel={({ value: resource }: SelectOption) => {
+                  const isSelected = userStore.hiddenMetrics.has(resource);
+
+                  return (
+                    <div className="flex gaps align-center">
+                      <span>{resource}</span>
+                      {isSelected && <Icon small material="check" className="box right"/>}
+                    </div>
+                  );
+                }}
+              />
+            </div>
+            <Checkbox
+              label="Hide metrics for all resources"
+              value={userStore.preferences.hideMetrics}
+              onChange={v => userStore.preferences.hideMetrics = v}
+            />
+          </WizardStep>
+        </Wizard>
+      </Dialog>
```

Comment:
Sorry for not noticing this sooner. In trying to find a better solution to the `min-height` problem on "re-showing" the metrics. I had to use this a lot and I personally find it very cumbersome.

Would you be able to remake this aspect of the PR more in line with how the rest of settings is set out:

1. No dialog or Wizard
2. Should be able to see which metrics are disabled
3. The title should be "Hide Metrics" or something to that affect
4. The placeholder text should correctly indicate what is happening
5. Maybe use something like what is used for Helm repos, which lets the user easily clear multiple items.

Reply:
Thank you for your review and time.  I will do my best to make it more usable. I've changed it completely. Wdyt? (I'm not sure that we have to show hidden metrics in placeholder, because they are placed below)
One more question. What do you think about `min-height` and grid row areas? I just want to fix all comments at once

<img width="915" alt="Screen Shot 2021-02-19 at 4 01 35 PM" src="https://user-images.githubusercontent.com/38247153/108502243-f0dee880-72cb-11eb-90eb-c60c79d12e21.png">
<img width="927" alt="Screen Shot 2021-02-19 at 4 04 37 PM" src="https://user-images.githubusercontent.com/38247153/108502395-2d124900-72cc-11eb-8012-2e39c6660b97.png">

Reply:
1. From my investigation the `min-height` solution works (and I wasn't able to figure out why it wasn't working) so lets leave it.
2. The named `grid-rows` is good
3. I agree that not having the names in the select is good.
4. The `X`'s beside the hidden metric names below the select seem to be slightly larger (taller) than the font size of the names. Could you make them the same?

Other than that looks good.

Reply:
Finally, after some discussion we think that this would be better as a cluster setting rather than a lens wide setting (since metrics might be available for some clusters but not for others), wdyt?

Reply:
Agree. I will move this feature to cluster settings